server/routers/viewer.tsx

```diff
@@ -0,0 +1,81 @@
+import { TRPCError } from "@trpc/server";
+
+import { createRouter } from "../createRouter";
+
+// routes only available to authenticated users
+export const viewerRouter = createRouter()
+  // check that user is authenticated
+  .middleware(({ ctx, next }) => {
+    const { user } = ctx;
+    if (!user) {
+      throw new TRPCError({ code: "UNAUTHORIZED" });
+    }
+
+    return next({
+      ctx: {
+        ...ctx,
+        // session value is known to be non-null now
+        user,
+      },
+    });
+  })
+  .query("me", {
+    resolve({ ctx }) {
+      return ctx.user;
+    },
+  })
+  .query("bookings", {
+    async resolve({ ctx }) {
+      const { prisma, user } = ctx;
+      const bookingsQuery = await prisma.booking.findMany({
+        where: {
+          OR: [
+            {
+              userId: user.id,
+            },
+            {
+              attendees: {
+                some: {
+                  email: user.email,
+                },
+              },
+            },
+          ],
+        },
+        select: {
+          uid: true,
+          title: true,
+          description: true,
+          attendees: true,
+          confirmed: true,
+          rejected: true,
+          id: true,
+          startTime: true,
+          endTime: true,
+          eventType: {
+            select: {
+              team: {
+                select: {
+                  name: true,
+                },
+              },
+            },
+          },
+          status: true,
+        },
+        orderBy: {
+          startTime: "asc",
+        },
+      });
+
+      const bookings = bookingsQuery.reverse().map((booking) => {
+        return {
+          ...booking,
+          startTime: booking.startTime.toISOString(),
```

Comment:
Is this necessary (tRPC has issues dealing with Date/Dayjs objects?)

Reply:
I just copy-pasted what was already in `getServerSideProps`.

In tRPC we could actually just use a `Date`-object if we add superjson.

Ref: https://trpc.io/docs/data-transformers